modules/aggs-matrix-stats/src/main/java/org/elasticsearch/search/aggregations/matrix/stats/MatrixStatsAggregator.java

```diff
@@ -83,7 +81,7 @@ public LeafBucketCollector getLeafCollector(LeafReaderContext ctx,
             public void collect(int doc, long bucket) throws IOException {
                 // get fields
                 if (includeDocument(doc)) {
-                    stats = bigArrays.grow(stats, bucket + 1);
+                    stats = bigArrays().grow(stats, bucket + 1);
```

Comment:
I expect this method call to get inlined. It is a chain of a few method calls, one of which is bimorphic, but the bimorphic call is on a `final` field so I think the compiler can inline that too. My guess is that this is going to end up compiled to a field access so running it once and putting it in a local isn't going to help.